### PR TITLE
[clang][cas] Move `CompileJobCache` into `Frontend` library

### PR DESCRIPTION
**Pull request by akyrtzi**

### CODE REVIEW

**Comment by akyrtzi** [comment]
@swift-ci Please test

File: clang/lib/Frontend/CompileJobCache.cpp [hunk 1]
```diff
@@ -0,0 +1,981 @@
...
+#include "clang/Basic/DiagnosticCAS.h"
+#include "clang/Frontend/CASDependencyCollector.h"
+#include "clang/Frontend/CompileJobCacheKey.h"
+#include "clang/Frontend/CompilerInstance.h"
+#include "clang/Lex/PreprocessorOptions.h"
+#include "llvm/CAS/ActionCache.h"
+#include "llvm/CAS/CASOutputBackend.h"
+#include "llvm/CASUtil/Utils.h"
+#include "llvm/MCCAS/MCCASObjectV1.h"
+#include "llvm/RemoteCachingService/Client.h"
```
Comment (benlangmuir) [inline clang/lib/Frontend/CompileJobCache.cpp:20]: I think we need to add `MCCAS` and `RemoteCachingService` as a target dependencies in `CMakeLists.txt`. We might also be able to remove those targets from `tools/driver/CMakeLists.txt ` now?

**Review by benlangmuir — COMMENTED** [review]
Mostly LGTM, though I didn't review the code in CompileJobCache.cpp in detail -- I assume it's mostly just moved? I saw you made CachingOutputs a nested type, which seems fine. Were there any interesting changes?

**Comment by akyrtzi** [comment]
> Mostly LGTM, though I didn't review the code in CompileJobCache.cpp in detail -- I assume it's mostly just moved? I saw you made CachingOutputs a nested type, which seems fine. Were there any interesting changes?

No, the other change other than moving was making these out-of-line:
```
CompileJobCache::CompileJobCache() = default;
CompileJobCache::~CompileJobCache() = default;

int CompileJobCache::reportCachingBackendError(DiagnosticsEngine &Diag,
                                               Error &&E) {
  Diag.Report(diag::err_caching_backend_fail) << llvm::toString(std::move(E));
  return 1;
}
```

**Comment by akyrtzi** [comment]
@swift-ci Please test

File: clang/lib/Frontend/CompileJobCache.cpp (hunk 1 above)
Comment (akyrtzi) [reply clang/lib/Frontend/CompileJobCache.cpp:20]: Done 👍

**Review by benlangmuir — APPROVED** [review]

**Pull request merged**